packages/react-utils/src/compose/resolveShorthandProps.tsx

```diff
@@ -6,14 +6,17 @@ import * as React from 'react';
  * @param props - The incoming props
  * @param shorthandPropNames - An array of prop names to apply simplification to
  */
-export const resolveShorthandProps = <TProps,>(props: TProps, shorthandPropNames: (keyof TProps)[]) => {
+export const resolveShorthandProps = <TProps,>(props: TProps, shorthandPropNames: string[]) => {
```

Comment:
yes it's a part of the public API, and the package is published... Do you have any ideas on how to avoid the breaking change ?

Reply:
```
 in next release -> when we'll fix it( revert back to enable only explicitly known set of keys)
```

But if we every type it to be more strict it will still be a breaking change right ? since right now because of `GenericDictionary` it is essentially a `string[]`

Reply:
One way to type it a bit more explicitly is to keep `(keyof TProps)[]` and then each component will need to explicitly type its array of shothand names to

`const componentSlots: (keyof ComponentProps)[] = [...]`

This would at least not let you declare any shorthand slots that aren't even in component props, but since we inherit the HTML attributes.... it's smth like 290+ keys

Reply:
To be clear this does not make anything worse. `string[]` is required now because previously any usages of this `TProps` actually extended `GenericDictionary`. 

Currently in master you can define any slots you anyway